app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -150,8 +152,29 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void onLongPress(MotionEvent e) {
-        // Also switch on long-press for accessibility
-        setMainImage(true, true);
+        openCurrentMainImageInGallery();
+    }
+
+    private void openCurrentMainImageInGallery() {
+        ImageType wantedImageType = imageTypes.get(mainImageIndex);
+
+        File file = null;
+        if (wantedImageType == ImageType.IMAGE_FRONT) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.front);
+        }
+        else if (wantedImageType == ImageType.IMAGE_BACK) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.back);
+        }
+
+        //Unreachable condition, but added this for safety
```

Comment:
This condition isn't unreachable, it is actually very common. If you remove this if check Catima will crash whenever you long-press a barcode.

```suggestion
        // Entry is either the barcode (not an image) or image retrieval failed
```